Doc/whatsnew/3.11.rst

```diff
@@ -752,6 +752,110 @@ Porting to Python 3.11
   which are not available in the limited C API.
   (Contributed by Victor Stinner in :issue:`46007`.)
 
+* Changes of the :c:type:`PyFrameObject` structure members:
+
+  * ``f_code``: removed, use :c:func:`PyFrame_GetCode` instead.
+    Warning: the function returns a :term:`strong reference`, need to call
+    :c:func:`Py_DECREF`.
+  * ``f_back``: changed, use :c:func:`PyFrame_GetBack`.
+  * ``f_builtins``: removed,
+    use ``PyObject_GetAttrString(frame, "f_builtins")``.
+  * ``f_globals``: removed,
+    use ``PyObject_GetAttrString(frame, "f_globals")``.
+  * ``f_locals``: removed,
+    use ``PyObject_GetAttrString(frame, "f_locals")``.
+  * ``f_lasti``: removed,
+    use ``PyObject_GetAttrString(frame, "f_lasti")``.
+  * ``f_valuesstack``: removed.
+  * ``f_stackdepth``: removed.
+  * ``f_gen``: removed.
+  * ``f_iblock``: removed.
+  * ``f_state``: removed.
+  * ``f_blockstack``: removed.
+  * ``f_localsplus``: removed.
+
+  The Python frame object is now created lazily. A side effect is that the
+  ``f_back`` member must not be accessed directly, since its value is now also
+  computed lazily. The :c:func:`PyFrame_GetBack` function must be called
+  instead.
```

Comment:
I created https://bugs.python.org/issue46356 to propose to enforce the usage of PyFrame_GetBack() on Python 3.11 to make the C API less error prone.